src/Raven.Server/Commercial/LetsEncryptUtils.cs

```diff
@@ -0,0 +1,968 @@
+
+                if (response.IsSuccessStatusCode == false)
+                {
+                    throw new InvalidOperationException(
+                        $"Got unsuccessful response from registration request: {response.StatusCode}.{Environment.NewLine}{responseString}");
+                }
+
+                if (parameters.Challenge == null)
+                {
+                    var existingSubDomain =
+                        registrationInfo.SubDomains.FirstOrDefault(x =>
+                            x.SubDomain.StartsWith(parameters.SetupInfo.LocalNodeTag + ".", StringComparison.OrdinalIgnoreCase));
+                    if (existingSubDomain != null &&
+                        new HashSet<string>(existingSubDomain.Ips).SetEquals(parameters.SetupInfo.NodeSetupInfos[parameters.SetupInfo.LocalNodeTag].Addresses))
+                    {
+                        parameters.Progress.AddInfo("DNS update started successfully, since current node (" + parameters.SetupInfo.LocalNodeTag + ") DNS record didn't change, not waiting for full DNS propagation.");
+                        return;
+                    }
+                }
+
+                var id = (JsonConvert.DeserializeObject<Dictionary<string, string>>(responseString) ?? throw new InvalidOperationException()).First().Value;
+
+                try
+                {
+                    RegistrationResult registrationResult;
+                    var i = 1;
+                    do
+                    {
+                        try
+                        {
+                            await Task.Delay(1000, cts.Token);
+                            response = await ApiHttpClient.Instance.PostAsync("api/v1/dns-n-cert/registration-result?id=" + id,
+                                    new StringContent(serializeObject, Encoding.UTF8, "application/json"), cts.Token)
+                                .ConfigureAwait(false);
+                        }
+                        catch (Exception e)
+                        {
+                            throw new InvalidOperationException("Registration-result request to api.ravendb.net failed.", e); //add the object we tried to send to error
+                        }
+
+                        responseString = await response.Content.ReadAsStringAsync(cts.Token).ConfigureAwait(false);
+
+                        if (response.IsSuccessStatusCode == false)
+                        {
+                            throw new InvalidOperationException(
+                                $"Got unsuccessful response from registration-result request: {response.StatusCode}.{Environment.NewLine}{responseString}");
+                        }
+
+                        registrationResult = JsonConvert.DeserializeObject<RegistrationResult>(responseString);
+                        if (i % 120 == 0)
+                            parameters.Progress.AddInfo("This is taking too long, you might want to abort and restart if this goes on like this...");
```

Comment:
Can you paste an example output of running the command? Does the user see all of those messages?